new.go

```diff
@@ -32,6 +32,7 @@ import (
 var NewSegmentBufferNumResultsBump int = 100
 var NewSegmentBufferNumResultsFactor float64 = 1.0
 var NewSegmentBufferAvgBytesPerDocFactor float64 = 1.0
+var MaxSegmentBufferInitialSize int = 10 * 1024 * 1024 // 10MB
```

Comment:
I'm thinking, does it make it sense it make this as a config option in bleve? If we have that, perhaps it would make things a bit more flexible and accommodative based on the use case of application, like the application can run some iterations with different values and set it as a config which works the best for their use case which need not be the optimal value for other use cases.